### DISCUSSION TITLE
git-tags datasource fails for Azure DevOps project names with whitespace

### DISCUSSION THREAD

**Discussion by alexanderlinne** [renovatebot/renovate#23056] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.147.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

The `git-tags` datasource fails to search for updates using a URL with whitespaces (e.g. `https://dev.azure.com/my-org/my%20project/_git/my-repo`). It works when there is no whitespace in the project name.
The problem appears to be, that the `git-url-parse` package decodes the URL but does not re-encode it. https://github.com/IonicaBizau/git-url-parse/pull/156 would be a fix for this problem, but the project does not seem to be actively maintained at the moment.
Given that this package is quite small, would you be open to inline this code with the fix or to use another implementation?

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Datasource unknown error (repository=my project/my-repo)
       "datasource": "git-tags",
       "packageName": "https://dev.azure.com/my-org/my%20project/_git/my-repo",
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "https://**redacted**@dev.azure.com/my-org/my project/_git/my-repo"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "fatal: unable to access 'https://dev.azure.com/my-org/my project/_git/my-repo/': URL rejected: Malformed input to a URL function\n",
         "stack": "Error: fatal: unable to access 'https://dev.azure.com/my-org/my project/_git/my-repo/': URL rejected: Malformed input to a URL function\n\n    at Object.action (/usr/local/lib/node_modules/renovate/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/usr/local/lib/node_modules/renovate/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /usr/local/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/local/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/usr/local/lib/node_modules/renovate/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/local/lib/node_modules/renovate/node_modules/simple-git/dist/cjs/index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
It would be better for us to fork the repo into this org and maintain the appropriate license/credits to the original author

**Reply by alexanderlinne** [reply]
Haven't thought about that, but would definitely be a better solution. I would be happy to create a PR with the fix in the fork and one to update the dependency in the main repo.

**Reply by rarkins** [reply]
@JamieMagee @viceice do you think we can use a git/GitHub dependency or better to publish to npm?

**Reply by viceice** [reply]
better publish to npm

**Reply by alexanderlinne** [reply, marked as answer]
@lfcyja now also opened #23354 which would solve the problem at the call site
